### PR TITLE
feat(semantic-release): implement automatic releases

### PR DESCRIPTION
**Pull request by dargmuesli**
Merge after https://github.com/fwalzel/handlebars-i18n/pull/22.
As I'm not a maintainer of this project, I cannot really check if the pipeline works. If it's ok for you, you could add me as a maintainer so I can make sure the pipeline is properly configured.

This PR shows how semantic-release could be implemented. This is an effort to lessen the time and thoughts needed for release management.
The way it works is the following:
- git commits need to be written in the Angular commit message format: you can take a look at my commits for an example. The format is automatically checked by the commitlint package and ran as husky hook automatically.
- depending on the commit message type a new release is created automatically and the maintainer (you) are free from creating releases manually, meanwhile giving all dependents frequently updated versions of this library. This makes it less likely to be having to react to security vulnerability notices too, as dependencies are already kept up to date by using renovate too. Renovate's updates are released once a week as patch level update, if no other updates occurred within the last 7 days.
- You will need to set two secrets in the GitHub project settings. One as `GITHUB_TOKEN` that's needed for tag push access to this repository. You can create this accordingly within your accounts developer settings under personal access tokens. The second token is called `NPM_TOKEN` and is what's currently used in the `.npmrc` file. You can set these tokens yourself and if you choose to add me as a maintainer, I would not be able to see them directly (I could upload a malicious GitHub workflow file, but I'm not interested in that ;)).

TL;DR
1. If you want me to configure and test this, you can add me as a maintainer. You can remove me after that's done too, that's ok.
2. You'd need to add the `NPM_TOKEN` secret in the GitHub project's settings.

If you want to keep the release process to yourself, feel free to discard this PR, no worries.

### CODE REVIEW

**Comment by dargmuesli** [comment]
Updated the PR's descriptions to explain the  `GITHUB_TOKEN` requirement (instead of `GH_TOKEN`) correctly.

**Comment by fwalzel** [comment]
I moved this repo to the freshly created "organization" named Aller-Couleur and sent you an invitation to join the team. As soon as you are a team member I grant you maintainer privileges.

**Comment by dargmuesli** [comment]
Joined! Thank you :)

**Comment by fwalzel** [comment]
Strangely, there are only two user roles available for your profile: owner and member. Git Documentation says, there would be also maintainer. Any ideas?

**Comment by dargmuesli** [comment]
Could you point me to the Git**Hub** documentation stating that?
I guess there are only `Owner` and `Member` roles on the organization level. There seems to be a [maintainer role **for teams**](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/giving-team-maintainer-permissions-to-an-organization-member). Of course I'd theoretically only need maintainer permissions for this specific project. You could add those permissions on the project's settings level or by giving the associated organization team the appropriate permissions.

**Comment by fwalzel** [comment]
I tried to follow this one: https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/giving-team-maintainer-permissions-to-an-organization-member

I created an organization and a team handlebars-i18n with us two as members … but no option maintainer available

**Comment by dargmuesli** [comment]
Ok, so you're not able to click on a checkbox next to my list entry over here and click on `change role...` from the dropdown menu above? https://github.com/orgs/Aller-Couleur/teams/handlebars-i18n/members

**Comment by fwalzel** [comment]
Funny enough, after logging in and out, the option is now available … :) You are maintainer now.

**Comment by dargmuesli** [comment]
Ok, thanks! I'm still missing privileges for this repository though. Maybe you can add it to the team over here https://github.com/orgs/Aller-Couleur/teams/handlebars-i18n/repositories or you can assign privileges directly in this repository's settings.

**Comment by fwalzel** [comment]
okay, the repo is now assign to our team … does it work for you now?

**Comment by dargmuesli** [comment]
Screenshot from 2021-04-26 10-18-47
not yet, maybe check point 8 here: https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/managing-team-access-to-an-organization-repository#giving-a-team-access-to-a-repository

**Comment by fwalzel** [comment]
Sorry for this super jerky on-boarding  :D  I have never administered this part of git yet. … the Team has now privileges on the Repo.

**Comment by dargmuesli** [comment]
Thanks!

**Comment by dargmuesli** [comment]
Ok, you'd now need to add the `NPM_TOKEN` secret in the GitHub project's settings.

**Comment by dargmuesli** [comment]
Just tell me when you've done this and I'll finish the setup.

**Comment by fwalzel** [comment]
Okay, I generated a GITHUB_TOKEN on the level User->Settings and the NPM_TOKEN on the level Repository->Secrets (There are also organizational and environmental secrets, but that's not what we need, right?).

Bildschirmfoto 2021-04-28 um 16 53 21


Bildschirmfoto 2021-04-28 um 16 45 27

**Comment by dargmuesli** [comment]
Looking good, except that the GITHUB_TOKEN, which needs write access to the repository, must be given as a repository secret too.
You create a access token with appropriate permissions, then you add the token as a secret to the repository so that the ci pipeline can make use of it and publish (write) releases on your behalf.

**Comment by fwalzel** [comment]
Repository secrets are not allowed to start with 'GITHUB_' (reserved word) as I just learned  ;)  so I named it SECRET_GITHUB_TOKEN. You should be able to access it via ${{ secrets.SECRET_GITHUB_TOKEN }}

**Comment by dargmuesli** [comment]
Hmm, it looks that the `GH_TOKEN` / `SECRET_GITHUB_TOKEN` isn't needed for this project. I now set the CI pipeline to use the default `GITHUB_TOKEN` for everything and it works too so far. Maybe the separate token was needed for private projects of mine, sorry for the confusion. Until further notice, you can remove the token from the repository's settings and your account again, I believe. I hope it stays that way now ;D
